metadata-ingestion/src/datahub/ingestion/source/mlflow.py

```diff
@@ -106,6 +109,21 @@ class MLflowConfig(StatefulIngestionConfigBase, EnvConfigMixin):
         ),
     )
 
+    class MaterializeDatasetInputsConfig(ConfigModel):
+        enabled: bool = Field(
+            default=False,
+            description="Whether to materialize dataset inputs for each run",
+        )
+
+        source_mapping_to_platform: Optional[dict] = Field(
+            default=None, description="Mapping of source type to datahub platform"
+        )
+
+    materialize_dataset_inputs: Optional[MaterializeDatasetInputsConfig] = Field(
+        default=MaterializeDatasetInputsConfig(),
+        description="Configuration for materializing dataset inputs for each run",
+    )
```

Comment:
```
config:
  materialize_dataset_inputs: true
  source_mapping_to_platform: {}
```

s3 dataset <- mlflow dataset <- ...

w/ materialize true:
- emit mlflow dataset w/ all entities
- including upstreamLineage pointing at s3
- also emit some aspects for the s3 dataset

w/ materialize false
- emit mlflow dataset w/ all entities
- including upstreamLineage pointing at s3 <- this still needs `source_mapping_to_platform`